cmd/sonobuoy/app/operations/cp.go

```diff
@@ -17,10 +17,138 @@ limitations under the License.
 package operations
 
 import (
+	"archive/tar"
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/kubectl/cmd"
 )
 
-func CopyResults(path string) error {
-	// pretty straight forwards
-	return errors.New("not implemented")
+// FileSpec defines what and where
+type FileSpec struct {
+	PodNamespace string
+	PodName      string
+	File         string
+}
+
+// CopyResults copies data from src to dst
+func CopyResults(config *rest.Config, clientset internalclientset.Interface, cmderr io.Writer, src, dest FileSpec) error {
+	reader, outStream := io.Pipe()
+	options := &cmd.ExecOptions{
+		StreamOptions: cmd.StreamOptions{
+			In:  nil,
+			Out: outStream,
+			Err: cmderr,
+
+			Namespace: src.PodNamespace,
+			PodName:   src.PodName,
+		},
+
+		// TODO: Improve error messages by first testing if 'tar' is present in the container?
+		Command:  []string{"tar", "cf", "-", src.File},
+		Executor: &cmd.DefaultRemoteExecutor{},
+	}
+
+	go func() {
+		options.Config = config
+		options.PodClient = clientset.Core()
+
+		defer outStream.Close()
+
+		if err := options.Validate(); err != nil {
+			fmt.Println("error validating")
```

Comment:
could we somehow bubble these errors up so we exit with nonzero if they don't succeed?

Reply:
these should probably go to stderr too